com.unity.shadergraph/Documentation~/Custom-Interpolators.md

```diff
@@ -0,0 +1,65 @@
+# Custom Interpolators
+
+## Description
+
+The Custom Interpolator feature provides fine-grained control over the specific calculations Shader Graph uses to bring data from the vertex stage to the pixel stage.
+
+There are two target audiences for Custom Interpolators:
+
+* Technical Directors and Lead Technical Artists setting up environments for their teams.
+* Graphics programmers helping artists to optimize content performance.
+
+
+## Supported data types
+Custom interpolators support float, vec2, vec3, and vec4 options.
+
+## Channel limits
+The Custom Interpolator feature supports a maximum of 32 channels. A channel is equivalent to four floats. Each float is an interpolator variable.
+Different platforms and GPUs have different interpolator variable limits. Exceeding the interpolator limitations of your target platform prevents your shaders from compiling. For detailed information about the number of interpolators supported by common interfaces, see the Unity documentation on [Shader semantics](https://docs.unity3d.com/Manual/SL-ShaderSemantics.html), and view the section **Interpolator count limits**. Test your Custom Interpolators on your target configuration to ensure that your content compiles properly.
+Technical directors can set warnings and errors to help their team members avoid creating graphs with too many channels to be compatible with their target pipeline, platform, or GPU. See **Creating channel warnings and errors** below.
+
+## How to use
+To use this feature, create a Custom Interpolator block in the Vertex context of the Master Stack and set a name and a data type. Create a vertex node to write data to that interpolator. Use the interpolator in your graph, then connect your graph to the relevant block in the Fragment context.
+These instructions include a contextual example illustrating the process of using a Custom Interpolator to fetch per-vertex data from a texture.
+To read the HLSL you use to replicate this behavior with the Built In Render Pipeline, see the Unity documentation on [Shader semantics](https://docs.unity3d.com/Manual/SL-ShaderSemantics.html) and view the section **Vertex ID: SV_VertexID**.
```

Comment:
@joshua-davis Does this sound right?

Reply:
@reallyfancy might be as good a person to answer as Joshua here (: